### ISSUE TITLE
Publish to AppSource fails

### ISSUE THREAD

**Issue by dsaveyn** [microsoft/AL-Go#1110]
It appears as if the solution for #1093 contains an issue.

When we deploy an app, we get the following error:

```
Error: Unexpected error when running action. Error Message: Unable to locate dependencies. You need to set generateDependencyArtifact to true in SharePoint_Integration/.AL-Go/settings.json in order to deliver dependencies to AppSource, StackTrace: at <ScriptBlock>, /home/runner/work/_actions/microsoft/AL-Go-Actions/v5.2/Deliver/Deliver.ps1: line 459 <- at <ScriptBlock>, /home/runner/work/_temp/8a679bf9-646e-471e-ae86-e97dbf9619d7.ps1: line 4 <- at <ScriptBlock>, <No file>: line 1
Error: Process completed with exit code 1.
```

Although the setting is configured on the project level:

```
{
  "country": "us",
  "appSourceCopMandatoryAffixes": [
    "FESC"
  ],
  "appFolders": [],
  "testFolders": [],
  "bcptTestFolders": [],
  "installApps": [
    "App/.appSourceCopPackages/*.app"
  ],
  "projectName": "SharePoint_Integration",
  "rulesetFile": "App/.vscode/app.ruleset.json",
  "deliverToAppSource": {
    "productId": "41f9d290-879e-48a7-a8ab-89f1382347f3",
    "mainAppFolder": "App",
    "continuousDelivery": false,
    "includeDependencies": [
      "*"
    ]
  },
  "generateDependencyArtifact": true
}
```

The settings for the Core App and on Repo Level are identical as documented in  #1093

These are the logs:
[Logs.txt](https://github.com/user-attachments/files/15902936/Logs.txt)

Any clue on what's wrong here?

**Comment by freddydk** [comment]
It looks like you are trying to publish rekease 23.0.18 - does that release contain the dependencies artifact?
This artifact is being uploaded if the setting is set when creating a release.
Like: 
![image](https://github.com/microsoft/AL-Go/assets/10775043/58206317-ff3c-476b-8c9c-49bdc0576dbb)

You can also publish the latest build to appsource by selecting latest when running the Publish to AppSource workflow.

**Comment by dsaveyn** [comment]
> It looks like you are trying to publish rekease 23.0.18 - does that release contain the dependencies artifact? 

I checked and no dependencies zip is included (in fact it isn't for any of our builds):

![image](https://github.com/microsoft/AL-Go/assets/92365904/f7af2285-0a5f-4ca2-97c6-a24f925799a0)

**GenerateDependencyArtifacts** is set on every of our projects, but not on repo level.

**Comment by freddydk** [comment]
I think it needs to be in the repo settings:
![image](https://github.com/microsoft/AL-Go/assets/10775043/d6d5ff51-da86-4610-af99-7b89786d5fc1)

**Comment by dsaveyn** [comment]
> I think it needs to be in the repo settings: ![image](https://private-user-images.githubusercontent.com/10775043/341705948-d6d5ff51-da86-4610-af99-7b89786d5fc1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5NjA1MDIsIm5iZiI6MTcxODk2MDIwMiwicGF0aCI6Ii8xMDc3NTA0My8zNDE3MDU5NDgtZDZkNWZmNTEtZGE4Ni00NjEwLWFmOTktN2I4OTc4NmQ1ZmMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIxVDA4NTY0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlMTBhNDM5MmYwM2EyOWFiMzcyYjE3OGZhYzJlNmEzMTYxYzc4MTcwMDA1YTMwMGU5ODYzODdmMzg4MmEwNzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nKEocdCkm6GUT8GYL35B36iwQdmAiFjMH9zfnmB22Ws)

Let me give that a try and get back to you

**Comment by dsaveyn** [comment]
Okay so I tried adding it the generateDependencyArtifact repo level again (it was there before) but no luck. I updated the system files to be sure as well.

I check the sample repo [here](https://github.com/microsoft/bcsamples-bingmaps.appsource/blob/main/.github/AL-Go-Settings.json) but there the setting is also only set on project level.

The settings on repo level:

```
{
  "type": "AppSource App",
  "templateUrl": "https://github.com/microsoft/AL-Go-AppSource@main",
  "enableCodeCop": true,
  "enableUICop": true,
  "rulesetFile": "",
  "enableExternalRulesets": true,
  "templateSha": "d3fa23ad4db0e18b7a5c602ab017659055ce45e2",
  "doNotRunTests": true,
  "doNotRunBcptTests": true,
  "useCompilerFolder": true,
  "doNotPublishApps": true,
  "githubrunner": "self-hosted, Windows",
  "githubRunnerShell": "powershell",
  "runs-on": "ubuntu-latest",
  "useProjectDependencies": true,
  "CICDPushBranches": [
    "main",
    "development",
    "release/*",
    "hotfix/*"
  ],
  "CICDPullRequestBranches": [
    "main",
    "development"
  ],
  "alwaysBuildAllProjects": false,
  "customCodeCops": [
    "https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.current.dll"
  ],
  "vsixFile": "latest",
  "features": [
    "TranslationFile"
  ],
  "repoVersion": "23.0",
  "versionStrategy": 23,
  "keyVaultName": "KV-AL-GO",
  "keyVaultcodesignCertificateName": "CodeSigningAlGo",
  "failOn": "warning",
  "artifact": "//23.1//latest",
  "generateDependencyArtifact": true
}
```

The settings on for the SharePoint Integration App (project level):

```
{
  "country": "us",
  "appSourceCopMandatoryAffixes": [
    "FESC"
  ],
  "appFolders": [],
  "testFolders": [],
  "bcptTestFolders": [],
  "installApps": [
    "App/.appSourceCopPackages/*.app"
  ],
  "projectName": "SharePoint_Integration",
  "rulesetFile": "App/.vscode/app.ruleset.json",
  "deliverToAppSource": {
    "productId": "41f9d290-879e-48a7-a8ab-89f1382347f3",
    "mainAppFolder": "App",
    "continuousDelivery": false,
    "includeDependencies": [
      "*"
    ]
  },
  "generateDependencyArtifact": true
}
```

The settings on for the Core App (project level) - dependency of the SharePoint Integration app:

```
{
  "country": "us",
  "appSourceCopMandatoryAffixes": [
    "ZESC"
  ],
  "appFolders": [],
  "testFolders": [],
  "bcptTestFolders": [],
  "installApps": ["App/.appSourceCopPackages/*.app"],
  "projectName": "Core",
  "rulesetFile": "App/.vscode/app.ruleset.json",
  "generateDependencyArtifact": true
}
```

Here's the log of the latest build:

[Logs_20240621.txt](https://github.com/user-attachments/files/15925656/Logs_20240621.txt)

These are the current artifacts:

![image](https://github.com/microsoft/AL-Go/assets/92365904/4a26f2cc-abce-4a51-8c51-29b60bd85618)

**Comment by freddydk** [comment]
The log only contains the initialization of the CI/CD job, I would need to see the full log.

**Comment by dsaveyn** [comment]
> The log only contains the initialization of the CI/CD job, I would need to see the full log.

Apologies, here they are:

[logs_25160130132.zip](https://github.com/user-attachments/files/15955416/logs_25160130132.zip)

**Comment by freddydk** [comment]
I think I know why this happens - will work on a fix.
I think it is related to UseCompilerFolder.

**Comment by freddydk** [comment]
I totally couldn't see the forest for all the trees.

The reason why this was failing is that you weren't using the preview of AL-Go for GitHub (which also was explained in the referenced bug #1093)

I have now released version 6.0.19 of BcContainerHelper, which includes the fix for the bug mentioned in that bug - which should make this work.

If you re-run, then the dependencies should be generated - but you might still need to use the preview of AL-Go for GitHub in order to publish to AppSource.

**Comment by dsaveyn** [comment]
> I totally couldn't see the forest for all the trees.
> 
> The reason why this was failing is that you weren't using the preview of AL-Go for GitHub (which also was explained in the referenced bug #1093)
> 
> I have now released version 6.0.19 of BcContainerHelper, which includes the fix for the bug mentioned in that bug - which should make this work.
> 
> If you re-run, then the dependencies should be generated - but you might still need to use the preview of AL-Go for GitHub in order to publish to AppSource.

The artifacts are now generated. Thank you! Any idea when this feature will be included in the main release of AL-Go?

**Comment by freddydk** [comment]
In the beginning of August
